File: samples/datastruct/java/structure.java
/* 
 * structure.java
 * Vino Crescini
 * 16 April 2001
 */
 
class structure
{
  private structure next;
  private structure prev;
  private int item;

  public structure() 
  {
    init();
  }
  public structure(int aitem, structure aprev, structure anext) 
  {
    init(aitem, aprev, anext);
  }
  public void init() 
  {
    prev = null;
    next = null;
    item = 0;
  }
  public void init(int aitem, structure aprev, structure anext) 
  {
    prev = aprev;
    next = anext;
    item = aitem;
  }
  public int getItem() 
  {
    return item;
  }
  public void setItem(int aitem) 
  {
    item = aitem;
  }
  public structure getNext() 
  {
    return next;
  }
  public void setNext(structure anext) 
  {
    next = anext;
  }
  public structure getPrev() 
  {
    return prev;
  }
  public void setPrev(structure aprev) 
  {
    prev = aprev;
  }
};

File: samples/datastruct/c/structure.cpp
/*
 * structure.cpp
 * Vino Crescini
 * 7 December 2000
 */

#include "structure.h"

structure::structure() 
{
  item = 0;
  next = NULL;
  prev = NULL;
}

structure::~structure()
{
}

void structure::setItem(int i) 
{
  item = i;
}

void structure::setNext(structure *s) 
{
  next = s;
}

void structure::setPrev(structure *s) 
{
  prev = s;
}

void structure::setStructure(int i, structure *n, structure *p) 
{
  setItem(i);
  setNext(n);
  setPrev(p);
}

int structure::getItem() 
{
  return item;
}
structure *structure::getNext() 
{
  return next;
}
structure *structure::getPrev() 
{
  return prev;
}

